source/elements/oneTBB/source/named_requirements/algorithms/par_for_each_body.rst

```diff
@@ -30,18 +30,31 @@ It should also meet one of the following requirements:
     ``ItemType`` may be optionally passed to ``Body::operator()`` by reference.
     ``const`` and ``volatile`` type qualifiers are also applicable.
 
-ItemType
---------
-
-The argument type ``ItemType`` should either satisfy the *CopyConstructibe*, *MoveConstructibe* or
-both requirements from the ISO C++ [utility.arg.requirements] section.
-If the type is not *CopyConstructibe*, there are additional usage restrictions:
-
-* If ``Body::operator()`` accepts an argument by value, or if the ``InputIterator`` type
-  from :doc:`parallel_for_each algorithm <../../algorithms/functions/parallel_for_each_func>` does
-  not satisfy the `Forward Iterator` requirements from the [forward.iterators] ISO C++ Standard section,
-  dereferencing an ``InputIterator`` must produce an rvalue reference.
-* Additional work items should be passed to the feeder as rvalues, for example, via the ``std::move`` function.
+Terms
+-----
+
+* ``iterator`` determines the type of the iterator passed into the ``parallel_for_each`` algorithm,
+  which is ``decltype(std::begin(c))`` for the overloads that accept the ``Container`` template argument or ``InputIterator``.
+* ``value_type`` - the type ``std::iterator_traits<iterator>::value_type``.
+* ``reference`` -  the type ``std::iterator_traits<iterator>::reference``.
+
+``tbb::parallel_for_each`` requires the ``Body::operator()`` call with an object of the ``reference`` type to be well-formed if
+the ``iterator`` meets all of the `Forward iterator` requirements described in the [forward.iterators] section of the 
+ISO C++Standard.
+
+`tbb::parallel_for_each algorithm <../../algorithms/functions/parallel_for_each_func>`
+requires the Body::operator() call with an object of type const value_type& or value_type&& to be well-formed if:
+
+* the iterator does not meet all of the `Forward iterator` requirements described in the [forward.iterators] section of the ISO C++ Standard
```

Comment:
What are we trying to say in this point? Could you please clarify the use-case?

Reply:
I agree that this case seems unclear, but here we are trying to describe the case when the iterator meets input iterator requirements but does not meet forward iterator requirements.

Reply:
It seems better for me, but I would swap the bullets:

... if all of the following requirements are met:
* The iterator meets the requirements of Input Iterator
* The iterator does not meet the requirements of Forward Iterator

As well, I would use the simple form "meets the requirements" instead of "meets all of the requirements". This "all of" is a little confusing for me

Reply:
@kboyarinov, @alexandraepan please review the update

Reply:
I am OK with the change